specs/Slots.md

```diff
@@ -0,0 +1,64 @@
+# Slots API specification
+
+Fluent UI components almost always contain sub parts and these pre-defined layouts can be configured via Slots API. Slots are named areas in a component that can receive props and provide styling and layout for them.
+
+## Slots definition
+
+For example, `Button` component contains `content`, `loader` and `icon` slots in its layout:
+
+```jsx
+<root>
+  <loader />
+  <content />
+  <icon />
+</root>
+```
+
+## Slots usage
+
+Slots can be configured through props, which lets the caller pass in a variety of inputs for a given slot. There are several forms of values that can be provided, but all of them share one common thing - they will produce a proper JSX layout inside a component.
+
+### An object as a value
+
+If you will pass an object to component props we will handle it as props for a slot. In an example below, props will be passed to an `icon` slot:
```

Comment:
I think we shold also mention that any attributes that are possible to use for standard JSX are fair game in shorthand

Reply:
it's a non-standard prop that we use quite frequently

Reply:
@ling1726 good suggestion 👍 I added an explicit section about this, is it clear now? Do you have any other suggestions?

Reply:
The extra code sample makes this a lot clearer, thanks.